### PR TITLE
Enable tests with atomic_fence_scope_capabilities and atomic_fence_order_capabilities

### PR DESCRIPTION
**Pull request by kopylovanat**
Enable tests with sycl::info::device::atomic_fence_scope_capabilities and sycl::info::device::atomic_fence_order_capabilities for DPCPP because issue https://github.com/intel/llvm/issues/8293 is resolved.

### CODE REVIEW

**Comment by kopylovanat** [comment]
DPCPP version need a bump for CI to pass.

File: tests/atomic_fence/atomic_fence.cpp [hunk 1]
```diff
@@ -74,17 +71,14 @@ bool check_atomic_fence_order_capability(sycl::queue& queue,
 bool check_atomic_fence_scope_capability(sycl::queue& queue,
                                          sycl::memory_scope scope,
                                          const std::string& scope_name) {
-// FIXME: re-enable when https://github.com/intel/llvm/issues/8293 is fixed
-#if !SYCL_CTS_COMPILING_WITH_DPCPP
   auto scopes =
       queue.get_device()
           .get_info<sycl::info::device::atomic_fence_scope_capabilities>();
-  if (std::find(scopes.begin(), scopes.end(), scope) == orders.end()) {
+  if (std::find(scopes.begin(), scopes.end(), scope) == scopes.end()) {
```
Comment (keryell) [inline tests/atomic_fence/atomic_fence.cpp:77]: Oops. Nice catch. :-)

File: tests/group_functions/group_barrier.cpp [hunk 2]
```diff
@@ -30,11 +30,9 @@ TEMPLATE_TEST_CASE_SIG("Group barriers", "[group_func][dim]", ((int D), D), 1,
                        2, 3) {
   auto queue = sycl_cts::util::get_cts_object::queue();
 
-  // FIXME: hipSYCL and DPCPP have no implemented
+  // FIXME: hipSYCL have no implemented
```
Comment (keryell) [inline tests/group_functions/group_barrier.cpp:33]:
```suggestion
  // FIXME: hipSYCL has not implemented
```

File: tests/group_functions/group_barrier.cpp [hunk 3]
```diff
@@ -44,7 +42,7 @@ TEMPLATE_TEST_CASE_SIG("Group barriers", "[group_func][dim]", ((int D), D), 1,
       sycl::memory_scope::sub_group, sycl::memory_scope::work_group,
       sycl::memory_scope::device, sycl::memory_scope::system};
   WARN(
-      "hipSYCL and DPCPP have no implementation of "
+      "hipSYCL have no implementation of "
```
Comment (keryell) [inline tests/group_functions/group_barrier.cpp:45]:
```suggestion
      "hipSYCL has no implementation of "
```

**Review by keryell — APPROVED** [review]
Thanks.
Just typos in messages.

**Pull request merged**